src/components/eRACommons.js

```diff
@@ -111,11 +106,31 @@ export const eRACommons = hh(class eRACommons extends React.Component {
 
   deleteNihAccount = async () => {
     AuthenticateNIH.eliminateAccount().then(
-      () => this.getUserInfo(),
+      async () => await this.getUpdatedUserInfo(),
       () => this.setState({ nihError: true })
     );
   };
 
+  getUpdatedUserInfo = async () => {
+    User.getMe().then((response) => {
+      const props = response.researcherProperties;
+      const authProp = find({'propertyKey':'eraAuthorized'})(props);
+      const expProp = find({'propertyKey':'eraExpiration'})(props);
+      const isAuthorized = isNil(authProp) ? false : getOr(false,'propertyValue')(authProp);
+      const expirationCount = isNil(expProp) ? 0 : AuthenticateNIH.expirationCount(getOr(0,'propertyValue')(expProp));
+      const nihValid = isAuthorized && expirationCount > 0;
+      const eraCommonsId = response.eraCommonsId;
+      this.props.onNihStatusUpdate(nihValid);
+      this.setState(prev => {
+        prev.currentUser = response;
+        prev.isAuthorized = isAuthorized;
+        prev.expirationCount = expirationCount;
+        prev.eraCommonsId = isNil(eraCommonsId) ? '' : eraCommonsId;
+        return prev;
+      });
```

Comment:
After having updated the user, we need to update where we store it. Add this line here:
```
      setUserRoleStatuses(response, Storage);
```
That method is in `utils.js` so you'll need an import. Also, we can rename the response to `user` or something that makes more sense.